Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM --platform=${BUILDPLATFORM:-linux/amd64} golang as build-env
+LABEL org.opencontainers.image.source=https://github.com/thatnerdjosh/go-example-webservices
+
+WORKDIR /app
+COPY . .
+ARG TARGETOS TARGETARCH
+ENV GOOS $TARGETOS
+ENV GOARCH $TARGETARCH
+RUN CGO_ENABLED=0 \
+    go build -ldflags="-s -w" -o task-webhook ./cmd/task-webhook/main.go
+
+# TODO: Extract to separate image
+RUN CGO_ENABLED=0 \
+    go build -ldflags="-s -w" -o auth ./cmd/auth/main.go
+
+FROM docker:cli
+WORKDIR /opt/go-example-webservices
+COPY ./docker-compose.yaml .
```

Comment:
Good point,  can volume mount that

Reply:
Not sure why your binary even needs compose, volume mounted or copied. Compose lives outside of you image

Reply:
The sample task I was given to implement was to restart using docker compose, how can I do that without the project setup and the yml?

Reply:
The task is just a generic webhook service, nothing specific here about docker or docker-compose. The provided example was just that - maybe not the best one. This thing is intended to run in the container because we like running services in containers. However, the actual hook will restart/execute things on the host. I.e. it can be "ssh hostaddress command" or "docker restart something". If we want to restart docker-compose on the host, it likely will do ssh to the host. Anyway, I get the source of the confusion, so it is clear to me why you have included compose to the image, and I agree - mapping volume is a better approach, however with the idea to restarting compose from within a running container, I can smell a lot of potential problems.

Reply:
Yup - I called out the need for orchestration in the Readme because that exact issue --- I spent a lot of time thinking through that portion of the task, didn't realize it meant any task x)